ci/clang-tidy-check/clang-tidy-check.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+# Copyright (c) 2020 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+set -e
+set -u
+set -o pipefail
+
+rc=0
+
+find_cmake_projects() {
+    find "$workdir" -name CMakeLists.txt -print0 | xargs -0 --no-run-if-empty grep -l 'option(RUN_CODE_CHECKS OFF)' | xargs dirname
```

Comment:
If we have `option(RUN_CODE_CHECKS ON)` the `clang-tidy` will not run and the script fails with `dirname: missing operand`. I think this is not the desired behavior.
Also, we need to add `--no-run-if-empty` for the last xargs command.

Reply:
799972a